packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2093,27 +2099,23 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      */
     private async createAndAliasDataStore(pkg: string | string[], alias: string, props?: any): Promise<IDataStore> {
         const internalId = uuid();
-        const dataStore = await this._createDataStore(pkg, false /* isRoot */, internalId, props);
-        const aliasedDataStore = channelToDataStore(dataStore, internalId, this, this.dataStores, this.mc.logger);
-        const result = await aliasedDataStore.trySetAlias(alias);
-        if (result !== "Success") {
-            throw new GenericError(
-                "dataStoreAliasFailure",
-                undefined /* error */,
-                {
-                    alias: {
-                        value: alias,
-                        tag: TelemetryDataTag.UserData,
-                    },
-                    internalId: {
-                        value: internalId,
-                        tag: TelemetryDataTag.PackageData,
-                    },
-                    aliasResult: result,
-                });
-        }
 
-        return aliasedDataStore;
+        try {
+            // A similar call may have been initiated by the same client, so we should try to get
+            // a possible existing aliased datastore first.
+            const existingDataStore = await this.getRootDataStoreChannel(alias, /* wait */ false);
+            return channelToDataStore(existingDataStore, internalId, this, this.dataStores, this.mc.logger, true);
+        } catch (err) {
```

Comment:
What kind of exception is thrown and where? If we go with this flow, we should find a way to use exceptions for exceptional conditions, not for control flow.

Reply:
Unfortunately the exception is part of the current protocol of getting root datastores. See `public async getDataStore(id: string, wait: boolean): Promise<FluidDataStoreContext>` inside `DataStores.ts`. I think changing that in this iteration is quite a heavy lift as it's a back-compat breaking change.